nixos/modules/services/networking/prosody.nix

```diff
@@ -91,36 +118,112 @@ let
       description = "Replies to XMPP pings with pongs";
     };
 
-    console = mkOption {
+    register = mkOption {
+      type = types.bool;
+      default = true;
+      description = "Allow users to register on this server using a client and change passwords";
+    };
```

Comment:
Was this previously built in, or were the users unable to register over XMPP?

Reply:
Ah, actually, this has not much to do with the 0.10 update in itself, I just stumbled over it during cleaning and creating the new module section. 😅

Previously, enabling this module was not exposed, but tied to the state of `services.prosody.allowRegistration`. The author probably thought that the module has no benefit when registration is disabled (it has, users can change their passwords themselves) or that enabling it would allow users to register even if `services.prosody.allowRegistration = false;` (it doesn't, this module is actually what gets configured through `services.prosody.allowRegistration`).

`services.prosody.allowRegistration = false;` and `services.prosody.modules.register = true;` is also upstream's default.

For server admins upgrading, the only effect should be that users can change their passwords via their XMPP clients now, even when public registration via XMPP is disabled, i.e. users are managed manually with the `prosodyctl` command line tool.